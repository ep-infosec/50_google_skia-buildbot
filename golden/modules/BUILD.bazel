load("//infra-sk:index.bzl", "karma_test", "sass_library", "ts_library")

karma_test(
    name = "common_test",
    src = "common_test.ts",
    deps = [
        ":common_ts_lib",
        "//golden/modules/search-controls-sk",
        "//infra-sk/modules:test_util_ts_lib",
        "@npm//@types/chai",
        "@npm//chai",
        "@npm//common-sk",
    ],
)

ts_library(
    name = "common_ts_lib",
    srcs = ["common.ts"],
    visibility = ["//visibility:public"],
    deps = [
        ":rpc_types_ts_lib",
        "//golden/modules/search-controls-sk",
        "@npm//common-sk",
    ],
)

ts_library(
    name = "demo_util_ts_lib",
    srcs = ["demo_util.ts"],
    visibility = ["//visibility:public"],
)

ts_library(
    name = "rpc_types_ts_lib",
    srcs = ["rpc_types.ts"],
    visibility = ["//visibility:public"],
)

ts_library(
    name = "settings_ts_lib",
    srcs = ["settings.ts"],
    visibility = ["//visibility:public"],
)

ts_library(
    name = "shared_demo_data_ts_lib",
    srcs = ["shared_demo_data.ts"],
    visibility = ["//visibility:public"],
    deps = ["@npm//common-sk"],
)

sass_library(
    name = "colors_sass_lib",
    srcs = ["colors.scss"],
    visibility = ["//visibility:public"],
)
