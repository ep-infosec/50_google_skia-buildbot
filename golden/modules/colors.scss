/* Common Gold Colors via CSS Variables.

 Can be imported via:

    @import '../colors';
*/

html {
  --gold: rgb(217, 95, 2);
  --bright-orange: rgb(255, 127, 0);

  --light-error-red: rgb(255, 230, 230);
  --transparent-white: rgba(255, 255, 255, 0);

  --triaged-positive: #1B9E77;
  --triaged-negative: #E7298A;
  --untriaged: #A6761D;
}
