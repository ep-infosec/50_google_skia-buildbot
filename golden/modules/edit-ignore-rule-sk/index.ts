import './edit-ignore-rule-sk';
