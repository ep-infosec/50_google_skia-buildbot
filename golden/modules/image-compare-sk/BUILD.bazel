load("//infra-sk:index.bzl", "karma_test", "sk_demo_page_server", "sk_element", "sk_element_puppeteer_test", "sk_page", "ts_library")
load("//golden/demo-page-assets:index.bzl", "GOLD_DEMO_PAGE_ASSETS")

sk_demo_page_server(
    name = "demo_page_server",
    sk_page = ":image-compare-sk-demo",
    static_assets = GOLD_DEMO_PAGE_ASSETS,
)

sk_element(
    name = "image-compare-sk",
    sass_deps = [
        "@npm//:node_modules/elements-sk/colors.scss",
    ],
    sass_srcs = ["image-compare-sk.scss"],
    sk_element_deps = ["//golden/modules/multi-zoom-sk"],
    ts_deps = [
        "//golden/modules:common_ts_lib",
        "//infra-sk/modules/ElementSk:index_ts_lib",
        "@npm//elements-sk",
        "@npm//lit-html",
    ],
    ts_srcs = [
        "image-compare-sk.ts",
        "index.ts",
    ],
    visibility = ["//visibility:public"],
)

sk_page(
    name = "image-compare-sk-demo",
    html_file = "image-compare-sk-demo.html",
    sk_element_deps = [":image-compare-sk"],
    ts_deps = ["@npm//common-sk"],
    ts_entry_point = "image-compare-sk-demo.ts",
)

ts_library(
    name = "image-compare-sk_po_ts_lib",
    srcs = ["image-compare-sk_po.ts"],
    visibility = ["//visibility:public"],
    deps = [
        "//golden/modules/multi-zoom-sk:multi-zoom-sk_po_ts_lib",
        "//infra-sk/modules/page_object:page_object_element_ts_lib",
        "//infra-sk/modules/page_object:page_object_ts_lib",
    ],
)

sk_element_puppeteer_test(
    name = "image-compare-sk_puppeteer_test",
    src = "image-compare-sk_puppeteer_test.ts",
    sk_demo_page_server = ":demo_page_server",
    deps = [
        ":image-compare-sk_po_ts_lib",
        "//puppeteer-tests:util_ts_lib",
        "@npm//@types/chai",
        "@npm//chai",
    ],
)

karma_test(
    name = "image-compare-sk_test",
    src = "image-compare-sk_test.ts",
    deps = [
        ":image-compare-sk",
        ":image-compare-sk_po_ts_lib",
        "//infra-sk/modules:test_util_ts_lib",
        "@npm//@types/chai",
        "@npm//chai",
    ],
)
