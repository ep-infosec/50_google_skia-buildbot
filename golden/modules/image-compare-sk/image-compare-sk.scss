@import 'node_modules/elements-sk/colors.scss';

image-compare-sk {
  min-width: 510px;
  display: block;

  .comparison_bar {
    display: flex;
    flex-direction: row;
  }

  figure {
    display: flex;
    flex-direction: column;
    margin: 0;
  }

  figcaption {
    .legend_dot {
      width: 8px;
      height: 8px;
      border-radius: 8px;
      background-color: var(--black);
      display: inline-block;
      vertical-align: middle;
    }
  }

  .no_compare, .computing {
    margin: auto;
    font-weight: bold;
  }

  .zoom_btn {
    width: 90%;
  }

  multi-zoom-sk {
    width: 930px;
  }

  img.thumbnail {
    margin: 5px;
    border: solid 2px lightgray;
    display: block;
    width: 128px;
    height: 128px;
    object-fit: scale-down;
    object-position: top left;
    cursor: zoom-in;
  }

  img.diff {
    image-rendering: pixelated;
  }

  img.fullsize {
    width: inherit;
    height: inherit;
    cursor: zoom-out;
  }
}
