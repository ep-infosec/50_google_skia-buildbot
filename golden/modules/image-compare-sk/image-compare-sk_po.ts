import { PageObject } from '../../../infra-sk/modules/page_object/page_object';
import { PageObjectElement, PageObjectElementList } from '../../../infra-sk/modules/page_object/page_object_element';
import { MultiZoomSkPO } from '../multi-zoom-sk/multi-zoom-sk_po';

/** A page object for the ImageCompareSk component. */
export class ImageCompareSkPO extends PageObject {
  get multiZoomSkPO(): MultiZoomSkPO {
    return this.poBySelector('multi-zoom-sk', MultiZoomSkPO);
  }

  private get zoomBtn(): PageObjectElement {
    return this.bySelector('button.zoom_btn');
  }

  private get closeZoomDialogBtn(): PageObjectElement {
    return this.bySelector('dialog.zoom_dialog button.close_btn');
  }

  private get images(): PageObjectElementList {
    return this.bySelectorAll('img');
  }

  private get imageAnchors(): PageObjectElementList {
    return this.bySelectorAll('figcaption a');
  }

  private get zoomDialog(): PageObjectElement {
    return this.bySelector('dialog.zoom_dialog');
  }

  getImageSrcs(): Promise<(string | null)[]> {
    return this.images.map((img) => img.getAttribute('src'));
  }

  getImageCaptionTexts(): Promise<string[]> {
    return this.imageAnchors.map((a) => a.innerText);
  }

  getImageCaptionHrefs(): Promise<(string | null)[]> {
    return this.imageAnchors.map((a) => a.getAttribute('href'));
  }

  async isZoomBtnVisible(): Promise<boolean> {
    return !(await this.zoomBtn.hasAttribute('hidden'));
  }

  async clickZoomBtn(): Promise<void> {
    await this.zoomBtn.click();
  }

  isZoomDialogVisible(): Promise<boolean> {
    return this.zoomDialog.hasAttribute('open');
  }

  async clickCloseZoomDialogBtn(): Promise<void> {
    await this.closeZoomDialogBtn.click();
  }

  async clickImage(index: number): Promise<void> {
    const image = await this.images.item(index);
    await image.click();
  }
}
