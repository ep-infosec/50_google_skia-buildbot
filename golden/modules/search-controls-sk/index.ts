import './search-controls-sk';

export { SearchCriteriaToHintableObject, SearchCriteriaFromHintableObject } from './search-controls-sk';
