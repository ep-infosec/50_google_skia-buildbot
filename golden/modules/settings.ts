/**
 * This file lists helpers for returning global constants in Gold. These should not change once
 * an instance has booted up, so they need not be returned via JSON requests.
 *
 * Settings are expected to be in window.GoldSettings and the functions here are nice helpers
 * for that, so as to demystify "where do these values come from?"
 */

export interface GoldSettings {
  title?: string;
  defaultCorpus?: string;
  baseRepoURL?: string;
}

function getSettings(): GoldSettings | undefined {
  return (window as any).GoldSettings as GoldSettings | undefined;
}

export function title(): string {
  return getSettings()?.title || '';
}

export function defaultCorpus(): string {
  return getSettings()?.defaultCorpus || '';
}

export function baseRepoURL(): string {
  return getSettings()?.baseRepoURL || '';
}

export function testOnlySetSettings(newSettings: GoldSettings) {
  (window as any).GoldSettings = newSettings;
}
