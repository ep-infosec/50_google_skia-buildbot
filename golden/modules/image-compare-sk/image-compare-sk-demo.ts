import './index';
import { $$ } from 'common-sk/modules/dom';
import { ImageCompareSk } from './image-compare-sk';

const aDigest = '6246b773851984c726cb2e1cb13510c2';
const bDigest = '99c58c7002073346ff55f446d47d6311';

let ele = new ImageCompareSk();
ele.left = {
  digest: aDigest,
  title: 'a digest title',
  detail: 'example.com#aDigest',
};
ele.right = {
  digest: bDigest,
  title: 'the other image',
  detail: 'example.com#bDigest',
};
$$('#normal')!.appendChild(ele);

ele = new ImageCompareSk();
ele.left = {
  digest: aDigest,
  title: 'a digest title',
  detail: 'example.com#aDigest',
};
$$('#no_right')!.appendChild(ele);

ele = new ImageCompareSk();
ele.left = {
  digest: aDigest,
  title: 'a digest title',
  detail: 'example.com#aDigest',
};
ele.right = {
  digest: bDigest,
  title: 'the other image',
  detail: 'example.com#bDigest',
};
ele.fullSizeImages = true;
$$('#full_size_images')!.appendChild(ele);
