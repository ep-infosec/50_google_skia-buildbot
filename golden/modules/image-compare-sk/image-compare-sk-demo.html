<!DOCTYPE html>
<html>
  <head>
    <title>image-compare-sk</title>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />

    <style>
      .container {
        width: 600px;
      }
    </style>
  </head>
  <body class="body-sk">
    <h2>digest-details-sk</h2>
    <div id="normal" class="container"></div>

    <h2>digest-details-sk no right</h2>
    <div id="no_right" class="container"></div>

    <h2>digest-details-sk full size images</h2>
    <div id="full_size_images" class="container"></div>
  </body>
</html>
